### ISSUE TITLE
Reference issue for deleted prop in signal handler

### ISSUE THREAD

**Issue by andrejcremoznik** [cferdinandi/reef#210]
https://github.com/cferdinandi/reef/blob/c8f8cc37e3467a1bf72315f404d145f9fa15a625/src/components/signal.js#L26-L30

If the goal is to return the deleted value with the emitter, you'll need to create a copy of it before deletion.

```js
		deleteProperty (obj, prop) {
			let value = structuredClone(obj[prop])
			delete obj[prop];
			emit(type, {prop, value, action: 'delete'});
			return true;
		}
```

**Comment by cferdinandi** [comment]
Good catch! All of my testing had been done with primatives as values, but this will absolutely not work when the property is an object (for mutation reasons you noted).

I'll get this done when I can, but if you wanted to submit a PR, that would probably move things along faster.

**Comment by TALlama** [comment]
This change breaks anything that holds a `Proxy` object, because they [cannot be cloned](https://developer.mozilla.org/en-US/docs/Web/API/Web_Workers_API/Structured_clone_algorithm#supported_types). Would it make sense to catch the resulting `DataCloneError` and fall back to using `obj[prop]` here without the cloning?

**Comment by cferdinandi** [comment]
@TALlama if I'm following your correctly, the returned property is no longer a reactive proxy, which is what you would expect it to be, yes?

I actually think that's ok, personally. The intent (in my mind) is to let you know what changed rather than provide a property that could trigger another reactive render.

I'm more than open to being shown why that's a bad idea, though!

**Comment by cferdinandi** [comment]
Fixed
